matchms/Fingerprints.py

```diff
@@ -0,0 +1,442 @@
+    fingerprint_type
+        Determine method for deriving molecular fingerprints. Supported choices are 'daylight',
+        'morgan1', 'morgan2', 'morgan3'.
+    nbits
+        Dimension or number of bits of generated fingerprint.
+
+    Returns
+    -------
+    fingerprint
+        Molecular fingerprint.
+    """
+    mol = Chem.MolFromSmiles(smiles)
+    if mol is None:
+        return None
+    return _mol_to_fingerprint(mol, fingerprint_algorithm, fingerprint_method, nbits, **kwargs)
+
+
+def _derive_fingerprint_from_inchi(inchi: str, fingerprint_algorithm: str, fingerprint_method: str, nbits: int, **kwargs) -> Optional[np.ndarray]:
+    """Calculate molecule fingerprint based on given inchi (using rdkit).
+    Requires conda package *rdkit* to be installed.
+
+    Parameters
+    ----------
+    inchi
+        Input InChI to derive fingerprint from.
+    fingerprint_type
+        Determine method for deriving molecular fingerprints. Supported choices are 'daylight',
+        'morgan1', 'morgan2', 'morgan3'.
+    nbits
+        Dimension or number of bits of generated fingerprint.
+
+    Returns
+    -------
+    fingerprint: np.array
+        Molecular fingerprint.
+    """
+    mol = Chem.MolFromInchi(inchi)
+    if mol is None:
+        return None
+    return _mol_to_fingerprint(mol, fingerprint_algorithm, fingerprint_method, nbits, **kwargs)
+
+
+def _mol_to_fingerprint(mol: Mol, fingerprint_algorithm: str, fingerprint_type: str, nbits: int, **kwargs) -> Optional[np.ndarray]:
+    """Convert rdkit mol (molecule) to molecular fingerprint.
+    Requires conda package *rdkit* to be installed.
+
+    Parameters
+    ----------
+    mol
+        Input rdkit molecule.
+    fingerprint_algorithm
+        Determine algorithm for deriving molecular fingerprints.
+        Supported algorithms are 'daylight', 'morgan1', 'morgan2', 'morgan3'.
+    fingerprint_type
+        Determine type for deriving molecular fingerprints.
+        Supported types are 'bit', 'sparse_bit', 'count', 'sparse_count'.
+    nbits
+        Dimension or number of bits of generated fingerprint.
+    **kwargs
+        Keyword arguments to pass additional parameters to FingerprintGenerator.
+        The keywords should match the corresponding RDKit implementation (e.g., min_path/max_path for RDKitFPGenerator).
+        See https://rdkit.org/docs/source/rdkit.Chem.rdFingerprintGenerator.html.
+
+    Returns
+    -------
+    fingerprint
+        Molecular fingerprint.
+    """
+    algorithms = {
+        "daylight": lambda args: GetRDKitFPGenerator(**args),
+        "morgan1": lambda args: GetMorganGenerator(**args, radius=1),
+        "morgan2": lambda args: GetMorganGenerator(**args, radius=2),
+        "morgan3": lambda args: GetMorganGenerator(**args, radius=3)
+    }
+    types = {
+        "bit": "GetFingerprint",
+        "sparse_bit": "GetSparseFingerprint",
+        "count": "GetCountFingerprint",
+        "sparse_count": "GetSparseCountFingerprint"
+    }
```

Comment:
This could all go to the beginning of the file, something like:
```python
FP_ALGORITHMS = {
    "daylight": lambda args: GetRDKitFPGenerator(**args),
    "morgan1": lambda args: GetMorganGenerator(**args, radius=1),
    "morgan2": lambda args: GetMorganGenerator(**args, radius=2),
    "morgan3": lambda args: GetMorganGenerator(**args, radius=3)
}

FP_SETTINGS = {
    "bit": "GetFingerprint",
    "sparse_bit": "GetSparseFingerprint",
    "count": "GetCountFingerprint",
    "sparse_count": "GetSparseCountFingerprint"
}
```

Reply:
(just to avoid duplication here)

Reply:
I can add the algorithms to the top of the file as a constant. Settings however, are different in _mol_to_fingerprint and _mols_to_fingerprints (mind the s in Settings)